### PR TITLE
Add pagination component

### PR DESCRIPTION
**Pull request by shekhar-amura**
Pagination component has added.

Screenshot from 2020-04-18 22-41-58

### CODE REVIEW

File: src/components/PrList.js [hunk 1]
```diff
@@ -1,10 +1,17 @@
...
 
 class PrList extends React.Component {
-  state = {pr_data: null, term: '', state: 'all', sort: 'asc', base: null }
+  constructor() {
+    super();
+    this.state = { pr_data: null, term: '', state: 'all', sort: 'asc', base: null, pageOfItems: [] }
+  }
 
+  onChangePage = (pageOfItems) => {
+    this.setState({ pageOfItems: pageOfItems });
```
Comment (tanmay7amura) [inline src/components/PrList.js:13]: Kindly use shorthand syntax

File: src/components/PrList.js [hunk 2]
```diff
@@ -1,10 +1,17 @@
 import React from 'react';
 import git from '../api/git';
+import Pagination from './Pagination';
 import './Style.css';
 
 class PrList extends React.Component {
-  state = {pr_data: null, term: '', state: 'all', sort: 'asc', base: null }
+  constructor() {
```
[the remark this answers is not in the document]
Comment (shekhar-amura) [reply src/components/PrList.js]: Yes. No need to add it.

File: src/components/PrList.js (hunk 1 above)
Comment (shekhar-amura) [reply src/components/PrList.js]: Done.

File: src/components/PrList.js [hunk 3]
```diff
@@ -82,12 +86,12 @@ class PrList extends React.Component {
               </thead>
               <tbody>
                 {pr_data_present
-                  ? (this.state.pr_data.map(function(object, i){
+                  ? (this.state.pageOfItems.map(function(object, i){
```
Comment (huzefa-amura) [inline src/components/PrList.js:89]: @shekhar-amura : From where does the value of `pageOfItems` gets set when the page is loaded initially?

Comment (shekhar-amura) [reply src/components/PrList.js:89]: we are updating pageOfItems from pagination component.  Once the pr_data is available then pagination component will pageOItems

Comment (huzefa-amura) [reply src/components/PrList.js:89]: @shekhar-amura : Okay! We need to make sure that this `pagination` component is re-usable for any other component too before a single line of code change.

**Comment by shekhar-amura** [comment]
@huzefa-amura yes it is reusable we just need to pass prop items to pagination component ahe add one state pageOfItems to the component. That's it.

**Pull request merged**